packages/@lwc/template-compiler/src/parser/constants.ts

```diff
@@ -168,10 +168,41 @@ export const KNOWN_HTML_ELEMENTS = new Set(HTML_ELEMENTS.concat(HTML_VOID_ELEMEN
 export const HTML_TAG = {
     A: 'a',
     AREA: 'area',
+    BODY: 'body',
+    CAPTION: 'caption',
+    COL: 'col',
+    COLGROUP: 'colgroup',
+    HEAD: 'head',
+    HTML: 'html',
+    TBODY: 'tbody',
+    TD: 'td',
+    TFOOT: 'tfoot',
+    TH: 'th',
+    THEAD: 'thead',
+    TR: 'tr',
     USE: 'use',
 };
 export const ATTR_NAME = {
     HREF: 'href',
     XLINK_HREF: 'xlink:href',
 };
 export const TEMPLATE_DIRECTIVES = [/^key$/, /^lwc:*/, /^if:*/, /^for:*/, /^iterator:*/];
+
+// Tags that cannot be parsed at the top level of a fragment. Generated from
+// https://github.com/sindresorhus/html-tags/blob/95dcdd5/index.js
+// using the test:
+// document.createRange().createContextualFragment(`<${tag}></${tag}>`).firstChild === null
```

Comment:
I think it's important to call out that we are doing this because we still support IE11. If it wasn't for IE11, we wouldn't need to use `Range.createContextFragment` and use the `<template>` element to parse fragments. 
Could we add a comment here and in `createFragment` under `@lwc/engine-dom` indicating that all of this should be dropped when we are moving away from IE11. 

Another option worth calling out would be to disable entirely static VNode hoisting in compat node and transition to `<template>` element for parsing fragments.
Pros:
- No need to maintain this special logic in the template compiler 
- Better performance for good citizens
Cons: 
- Support for 2 template compiler outputs at runtime

At this stage of the release, I would favor the approach with the least amount of changes (the current PR).

Reply:
Good call. I'll add the comment.

I looked back at the previous perf analysis I did (https://github.com/salesforce/lwc/pull/2781#issuecomment-1100280147), and it looks like using a `<template>` and setting `innerHTML` on it is just as fast as `createContextualFragment`. So yeah, we could switch to using that, if not for IE11.

Reply:
Opened #2890